src/cider/nrepl.clj

```diff
@@ -1,53 +1,398 @@
 (ns cider.nrepl
-  (:require [clojure.tools.nrepl.server :as nrepl-server]
-            [cider.nrepl.print-method]
-            [cider.nrepl.middleware.apropos]
-            [cider.nrepl.middleware.classpath]
-            [cider.nrepl.middleware.complete]
-            [cider.nrepl.middleware.debug]
-            [cider.nrepl.middleware.enlighten]
-            [cider.nrepl.middleware.format]
-            [cider.nrepl.middleware.info]
-            [cider.nrepl.middleware.inspect]
-            [cider.nrepl.middleware.macroexpand]
-            [cider.nrepl.middleware.ns]
-            [cider.nrepl.middleware.spec]
-            [cider.nrepl.middleware.out]
-            [cider.nrepl.middleware.pprint]
-            [cider.nrepl.middleware.refresh]
-            [cider.nrepl.middleware.resource]
-            [cider.nrepl.middleware.stacktrace]
-            [cider.nrepl.middleware.test]
-            [cider.nrepl.middleware.trace]
-            [cider.nrepl.middleware.track-state]
-            [cider.nrepl.middleware.undef]
-            [cider.nrepl.middleware.version]))
+  (:require [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.middleware.session :refer [session]]
+            [clojure.tools.nrepl.middleware.pr-values :refer [pr-values]]
+            [clojure.tools.nrepl.server :as nrepl-server]
+            [cider.nrepl.middleware.util.cljs :as cljs]
+            [cider.nrepl.middleware.pprint :as pprint]
+            [cider.nrepl.middleware.version]
+            [cider.nrepl.print-method]))
+
+(def DELAYS (atom nil))
+
+(defn- resolve-or-fail [sym]
+  (or (resolve sym)
+      (throw (IllegalArgumentException. (format "Cannot resolve %s" sym)))))
+
+(defmacro run-delayed-handler
+  "Make a delay of `fn-name` and place it in `DELAYS` atom at compile time.
+  Require and invoke the delay at run-time with arguments h and msg."
```

Comment:
What are h and msg? 😃

Reply:
This is an internal function. Definition of h and msg are in def-wrapper, if not it should be improved. Will smooth this out.